docs/enhancements/multi-rhel-fips.md

```diff
@@ -0,0 +1,244 @@
+  difficult to find out from the SaaS how many installer processes are being run
+  concurrently (maybe we should create a metric for this), but the telco scale
+  team regularly runs several hundred concurrently maxing out at over three
+  thousand in a few hours. Unless we're cleaning up the jobs rather aggressively
+  I don't think it would be good to create this many.
+
+- Multiple jobs would need to be run on a single assets directory. This seems
+  prohibitively complex compared to the proposed solution. During a single
+  install the following installer commands are used:
+  - `openshift-baremetal-install create manifests`
+  - `openshift-baremetal-install create single-node-ignition-config` or
+    `openshift-baremetal-install create ignition-configs` (depending on HA mode)
+
+### Run the matching installer on the assisted-service container
+
+Clusters that have installers that already match the assisted service container's
+architecture could be handled by the assisted-service container as we do today.
+This would require one less image and container per pod, but having the same
+process for every cluster install would be easier to understand and maintain.
+
+### Use RPC over HTTP
+
+[Go's RPC](https://pkg.go.dev/net/rpc@go1.22.3) could be used instead of a direct
+HTTP server (RPC can be hosted over HTTP, but that's not what is being addressed
+here). RPC would make this a simpler change as the code for generating the
+manifests is already contained in a single package, but RPC would be a strange
+choice if we were to move the handling into a truly separate service in the
+future.
+
+### Install multiple libraries on the same image
+
+It may be possible to install both versions of the shared libraries required by
+the installers (libcrypto and libssl?) for FIPS compliance on a single image.
+This would require much less change and should be significantly quicker to
+implement, but it's not clear if these would be possible or supportable.
+This could be achieved by any of the following methods:
+
+1. Create a separate userspace for el8 libraries and chroot when those libraries
+   are required.
+  - This seems a bit complicated and it will likely make our image quite a bit
+    larger than it already is (~1.3G).
+2. Install both versions of the required packages simultaneously.
+  - Not sure if this is possible given that the packages share a name and are
+    only different in version.
+3. Use multi-layer container builds to copy the libraries from an el8 image to a
+   directory on the el9 image and use `LD_PRELOAD` or manipulate `LD_LIBRARY_PATH`
+   to point the el8 installer binaries to the correct libraries.
+
+The approach using chroot worked, but FIPS SMEs said that the container base
+image *must* match the installer for the resulting cluster to be considered
+FIPS-compliant so none of these multi-library options are valid.
+
+### Publish multiple assisted-service images
```

Comment:
@romfreiman does this look like what you were talking about?
Tried to capture it here.